addons/api/addon/serializers/credential-library.js

```diff
@@ -31,9 +32,39 @@ export default class CredentialLibrarySerializer extends ApplicationSerializer {
         return super.serialize(...arguments);
     }
   }
+  serializeAttribute(snapshot, json) {
+    const value = super.serializeAttribute(...arguments);
+    const { credential_type, type, credential_mapping_overrides, isNew } =
+      snapshot.record;
+    if (
+      type === TYPE_CREDENTIAL_LIBRARY_VAULT_GENERIC &&
+      credential_mapping_overrides &&
+      credential_type &&
+      !isNew
+    ) {
+      //API expects to send null to fields if it is undefined or deleted
+      const remainingAttrs = Object.entries(
```

Comment:
Why is this needed?

Reply:
When updating the select list, the API explicitly wants us to send null to fields we don't need. Otherwise, they will return the unmodified value from previous versions. 

For ex, If I have a username attr and want to change that to a `password` attr, I will need to send both the fields and make the username null. Otherwise, they will send values for both fields.

Reply:
I had a setup where I changed the key and it wasn't setting null, just not sending the field which was working fine, is this expected?

https://github.com/hashicorp/boundary-ui/assets/5783847/e53f8d24-fca3-48b3-8268-6d07eebd7b38

Reply:
It is set in the serializer, so I don't think you would see it here?   I can see the null being sent to the API in the network call

Reply:
I think I understand now, I don't see it still but it doesn't matter from the API perspective. This was just ember re-using old values bc it assumes they're still relevant if we don't tell it otherwise